### PR TITLE
Skip preload download if --image-repository is set

### LINKED ISSUE #7612: Skip preload download if `--image-repository` is set
**重现问题所需的命令**：minikube start --cpus=2 --memory='2000mb' --vm-driver hyperkit --image-mirror-country='cn' --image-repository='registry.cn-hangzhou.aliyuncs.com/google_containers'

**失败的命令的完整输出**：<details>
😄  minikube v1.9.2 on Darwin 10.15.4
✨  Using the hyperkit driver based on existing profile
✅  Using image repository registry.cn-hangzhou.aliyuncs.com/google_containers
👍  Starting control plane node m01 in cluster minikube
💾  Downloading Kubernetes v1.18.0 preload ...
    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 131.59 MiB
</details>

**`minikube logs`命令的输出**： <details>

</details>

**使用的操作系统版本**：macOS 10.15.4

### PR DESCRIPTION
**Pull request by prasadkatti**
fixes #7612

### CODE REVIEW

**Comment by medyagh** [comment]
I dont agree with this PR ! I think if someone is in china we should try to download the preload from github.
we should instead try to upload our preload images to other than GCS bucket but also github

File: pkg/minikube/node/cache.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ const (
 
 // BeginCacheKubernetesImages caches images required for kubernetes version in the background
 func beginCacheKubernetesImages(g *errgroup.Group, imageRepository string, k8sVersion string, cRuntime string) {
-	if download.PreloadExists(k8sVersion, cRuntime) {
+	if imageRepository == "" && download.PreloadExists(k8sVersion, cRuntime) {
```
Comment (tstromberg) [inline pkg/minikube/node/cache.go:46]: Add:

TODO: remove imageRepository check once https://github.com/kubernetes/minikube/issues/7695 is fixed.

**Comment by tstromberg** [comment]
> I dont agree with this PR ! I think if someone is in china we should try to download the preload from github.
> we should instead try to upload our preload images to other than GCS bucket but also github

Yes, and this PR is to address the short-term bleeding for minikube v1.10 - as it's unlikely that we will have images elsewhere before then. Preload leaves man users in China in a broken state, causing an influx of incoming support issues. 

Additional thought: If a user sets --image-repository, they want the images to be loaded from a particular location, preloaded or not.  Even if we added GitHub, we should still allow a way to respect user preferences.

File: pkg/minikube/node/cache.go (hunk 1 above)
Comment (prasadkatti) [reply pkg/minikube/node/cache.go:47]: Thanks for the review. Added a TODO note.

**Comment by prasadkatti** [comment]
@tstromberg - Added a TODO note as a comment based on your review feedback.

**Pull request merged**